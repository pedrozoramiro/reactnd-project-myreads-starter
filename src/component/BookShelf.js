import React, { Component } from "react";

import Book from "./Book";
import PropTypes from "prop-types";

class BookShelf extends Component {
  render() {
    const { title, books, shelves, shelf, onChangeShelve } = this.props;
    //TODO: externalizar essa regra.
    const booksFilteredByShelf = books.filter(book => book.shelf === shelf);

    return (
      <div className="bookshelf">
        <h2 className="bookshelf-title">{title}</h2>
        <div className="bookshelf-books">
          <ol className="books-grid">
            {booksFilteredByShelf.map(function(book) {
              return (
                <li key={book.id}>
                  <Book
                    book={book}
                    shelves={shelves}
                    onChangeShelve={onChangeShelve}
                  />
                </li>
              );
            })}
          </ol>
        </div>
      </div>
    );
  }
}

BookShelf.propTypes = {
  title: PropTypes.string.isRequired,
  shelf: PropTypes.string.isRequired,
  shelves: PropTypes.array.isRequired,
  books: PropTypes.array.isRequired
};

export default BookShelf;
