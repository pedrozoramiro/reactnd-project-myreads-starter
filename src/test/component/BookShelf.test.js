import React from "react";
import { shallow, mount } from "enzyme";

import BookShelf from "../../component/BookShelf";


const mock = {
  books:[
    {
      id:'1',
      title: "Book [read]",
      shelf: "read",
      imageLinks: { smallThumbnail: "" },
      authors: ["Ramiro Pedrozo"]
    },
    {
      id:'2',
      title: "Book [wantToRead]",
      shelf: "wantToRead",
      imageLinks: { smallThumbnail: "" },
      authors: ["Ramiro Pedrozo"]
    },
    {
      id:'3',
      title: "Book [currentlyReading]",
      shelf: "currentlyReading",
      imageLinks: { smallThumbnail: "" },
      authors: ["Ramiro Pedrozo"]
    }
  ] ,
  shelves: [
    { name: "currentlyReading", title: "Currently Reading" },
    { name: "wantToRead", title: "Want to Read" },
    { name: "read", title: "Read" }
  ],
  handleChangeShelf: jest.fn(),
  shelf: {name:'wantToRead',title:'Want to Read'}
};

it("renders without crashing", () => {
  const bookShelfWrapper = (
    <BookShelf
      books={mock.books}
      shelf={mock.shelf}
      shelves={mock.shelves}
      handleChangeShelf={mock.handleChangeShelf}
    />
  );
  expect(shallow(bookShelfWrapper)).toMatchSnapshot();
});
