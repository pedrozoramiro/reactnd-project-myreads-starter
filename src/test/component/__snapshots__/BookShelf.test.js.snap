// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="bookshelf"
>
  <h2
    className="bookshelf-title"
  >
    Want to Read
  </h2>
  <div
    className="bookshelf-books"
  >
    <ol
      className="books-grid"
    >
      <li
        key="2"
      >
        <Book
          book={
            Object {
              "authors": Array [
                "Ramiro Pedrozo",
              ],
              "id": "2",
              "imageLinks": Object {
                "smallThumbnail": "",
              },
              "shelf": "wantToRead",
              "title": "Book [wantToRead]",
            }
          }
          handleChangeShelf={[Function]}
          shelves={
            Array [
              Object {
                "name": "currentlyReading",
                "title": "Currently Reading",
              },
              Object {
                "name": "wantToRead",
                "title": "Want to Read",
              },
              Object {
                "name": "read",
                "title": "Read",
              },
            ]
          }
        />
      </li>
    </ol>
  </div>
</div>
`;
